pages/hud.coffee

```diff
@@ -1,3 +1,21 @@
+document.addEventListener "keydown", (event) ->
```

Comment:
This is in a different frame to `FindMode`. Instead, they interact via `UIComponent`, which forwards the events to `FindMode` via `handlerStack.bubbleEvent` (which you commented on above). Most of the `FindMode` `keydown` code will probably end up in here, but not yet, for the sake of a small change footprint.

> why are we only handling `keydown`, here?

We only handle backspace, delete, up, down, enter and escape here. There's an `input` handler a little way down, which handles all the actual text, and is locale-agnostic because it uses `input`.